adbe/adb_enhanced.py

```diff
@@ -1602,3 +1607,119 @@ def print_notifications():
         for action in notification_actions:
             print_message('Action: %s' % action)
         print_message('')
+
+
+# Alarm Enum
+class AlarmEnum(Enum):
+    TOP = 't'
+    PENDING = 'p'
+    HISTORY = 'h'
+    ALL = 'a'
+
+
+# Prettify alarm logs
+def alarm_manager(param):
+    cmd = "dumpsys alarm"
+    c, o, e = execute_adb_shell_command2(cmd)
+
+    if c != 0:
+        print_error_and_exit("Something gone wrong on "
+                             "dumping alarms. Error: %s" % e)
+        return o
+
+    if isinstance(param, AlarmEnum):
+        run_all = 0
+        padding = ""
+
+        if param == AlarmEnum.ALL:
+            run_all = 1
+            padding = "   "
+
+        if param == AlarmEnum.TOP or run_all == 1:
+            print("Top Alarms:")
+            pattern_top_alarm = re.compile(r'(?<=Top Alarms:\n).*?(?=Alarm Stats:)',
+                                           re.DOTALL)
+            alarm_to_parse = re.sub(r' +', ' ',
+                                    re.search(pattern_top_alarm, o).group(0)).split("\n")
+            temp_dict = dict()
+
+            for i in range(len(alarm_to_parse)):
+                if re.match(r"^\+", alarm_to_parse[i]):
+                    temp_dict[alarm_to_parse[i]] = alarm_to_parse[i + 1]
+                    i += 1
+
+            for key, value in temp_dict.items():
+                # key example: +2m19s468ms running, 0 wakeups, 708 alarms: 1000:android
+                # value example: *alarm*:com.android.server.action.NETWORK_STATS_POLL
+                temp = key.split(',')
+                running_time = temp[0].split(" ")[0]
+                nb_woke_up = temp[1].strip().split(" ")[0]
+                nb_alarms = temp[2].strip().split(" ")[0]
+                uid = temp[2].strip().split(":")[1].strip()
+                package_name = temp[2].strip().split(":")[2].strip()
+
+                action = value.split(":")[1]
+
+                print("%sPackage name: %s" % (padding, package_name))
+                print("%s   Action: %s" % (padding * 2, action))
+                print("%s   Running time: %s" % (padding * 2, running_time))
+                print("%s   Number of device woke up: %s" % (padding * 2, nb_woke_up))
+                print("%s   Number of alarms: %s" % (padding * 2, nb_alarms))
+                print("%s   User ID: %s" % (padding * 2, uid))
+
+        if param == AlarmEnum.PENDING or run_all == 1:
+            print("Pending Alarms:")
```

Comment:
These if branches are long enough that you can extract functions out of them.

Reply:
Will update it during the week.